emulators/hyperkit/Portfile

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+
+name                hyperkit
+version             0.20210107
+categories          emulators
+platforms           darwin
+maintainers         nomaintainer
+license             BSD
+
+description         HyperKit is a toolkit for embedding hypervisor capabilities in your application.
+
+long_description    HyperKit is a toolkit for embedding hypervisor capabilities in your application. \
+                    It includes a complete hypervisor, based on xhyve/bhyve, which is optimized for \
+                    lightweight virtual machines and container deployment. It is designed to be interfaced \
+                    with higher-level components such as the VPNKit and DataKit.
+
+
+
+homepage            https://github.com/moby/hyperkit
+master_sites        https://github.com/moby/hyperkit/archive/refs/tags/
+
+distfiles           v${version}.tar.gz
+
+checksums           rmd160  49a397d86d20a7ea5d09261f3217b634d2a8b712 \
+                    sha256  095f5f5ef550d7cad10e4d13e9c9ce8b58cc319d654a6d837d8d87ee70537835 \
+                    size    1176356
+
+depends_build       port:ocaml \
+                    port:opam \
+                    port:pkgconfig \
+                    port:aspcud
+
+depends_lib         port:libev
+
+                    # ref: https://github.com/moby/hyperkit/issues/323
+patchfiles          implicit-conversion.patch
+
+use_configure no
+
+build {
+    set opam_libraries "uri \
+                        qcow.0.11.0 \
+                        conduit.2.1.0 \
+                        lwt.5.3.0 \
+                        qcow-tool \
+                        mirage-block-unix.2.12.0 \
+                        conf-libev \
+                        logs \
+                        fmt \
+                        mirage-unix \
+                        prometheus-app"
+
+    set opam_env_file ${workpath}/opam.env
+    set opam_pin ". ${opam_env_file} && opam pin add"
+    set opam_install ". ${opam_env_file} && opam install -j ${build.jobs} -y"
+
+    system -W ${workpath} "opam init -n --disable-sandboxing"
+    system "opam env > ${opam_env_file}"
+
+    system "${opam_pin} qcow.0.11.0 git://github.com/mirage/ocaml-qcow -n"
+    system "${opam_pin} qcow-tool.0.11.0 git://github.com/mirage/ocaml-qcow -n"
```

Comment:
is this downloading stuff in the `build` phase? If so, that's not ideal... but I must admit that I don't know anything about `opam` / `ocaml` so don't know whether that can be avoided.

Reply:
Sadly yes, it seems to be the norm with ocaml builds. I based a bit of this on the `semgrep` Portfile too.

Reply:
MacPorts has an ocaml portgroup that is intended to assist and simplify ports that install ocaml-based software. Does that help in this case? Read the comments at the top of the portgroup to find out how to use it. I'm surprised not to see anything in the portgroup related to using the right flags and compiler, so either the portgroup relies on existing MacPorts base code to do it or the portgroup may need to be improved to handle this.

Reply:
If necessary I can help make this independent of opam qua opam. Using an opam build for an ocaml package is not very compatible with the way macports works.